File: Laboratorio6/Libreria.py
class Libro:
    def __init__(self, Titulo, Autor, ISBN, Estado):
        self.__Titulo = Titulo
        self.__Autor = Autor
        self.__ISBN = ISBN
        self.__Estado = Estado
        pass

    def setTitulo(self, n_Titulo):
        self.__Titulo = n_Titulo
    
    def setAutor(self, n_Autor):
        self.__Autor = n_Autor

    def setISBN(self, n_ISBN):
        self.__ISBN = n_ISBN

    def setEstado(self, n_Estado):
        self.__Estado = n_Estado

    def getTitulo(self):
        return self.__Titulo
    
    def getAutor(self):
        return self.__Autor
    
    def getISBN(self):
        return self.__ISBN
    
    def getEstado(self):
        return self.__Estado
    
    def getInfo(self):
        return f"El titulo del libro: {self.getTitulo()} \nDel autor: {self.getAutor()} \nCon ISBN: {self.getISBN()} \nTiene el estado: {self.getEstado()}"
    
class Biblioteca:
    def __init__(self):
        self.__Libros = []
        pass

    def agregarLibro(self, n_Libro):
        self.__Libros.append(n_Libro)

    def buscarLibro(self, n_ISBN):
        for libro in self.__Libros:
            if libro.getISBN() == n_ISBN:
                return libro.getInfo()
        return "No existe el libro"

    def prestarLibro(self, n_ISBN):
        for libro in self.__Libros:
            if libro.getISBN() == n_ISBN:
                if libro.getEstado() == "Disponible":
                    libro.setEstado("Prestado")
                    return "Libro prestado con exito"
                else:
                    return "Este libro ya está prestado"
        return "No existe ese libro"
    
    def devolverLibro(self, n_ISBN):
        for libro in self.__Libros:
            if libro.getISBN() == n_ISBN:
                if libro.getEstado() == "Prestado":
                    libro.setEstado("Disponible")
                    return "Libro devuelto con exito"
                else:
                    return "Este libro ya está disponible"
        return "No existe ese libro"
    
    def mostrarLibros(self):
        if not self.__Libros:
            return "No hay libros en la biblioteca."

        info_libros = ""
        for libro in self.__Libros:
            info_libros += libro.getInfo() + "\n" + "-"*30 + "\n\n"
        return info_libros

if __name__ == '__main__':
    L1 = Libro("La divina comedia", "Dante Alighieri", "1", "Disponible")
    L2 = Libro("Don quijote de la mancha", "Miguel de Cervantes", "2", "Prestado")
    L3 = Libro("El paraíso perdido", "John Milton", "3", "Disponible")

    Biblio = Biblioteca()
    Biblio.agregarLibro(L1)
    Biblio.agregarLibro(L2)
    Biblio.agregarLibro(L3)
    print(Biblio.mostrarLibros())
    Biblio.prestarLibro("1")
    print(Biblio.mostrarLibros())
    Biblio.devolverLibro("2")
    print(Biblio.mostrarLibros())

File: Laboratorio6/Libreria.java
import java.util.ArrayList;

class Libro {
    private String titulo;
    private String autor;
    private String isbn;
    private String estado;

    public Libro(String titulo, String autor, String isbn, String estado) {
        this.titulo = titulo;
        this.autor = autor;
        this.isbn = isbn;
        this.estado = estado;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public void setEstado(String estado) {
        this.estado = estado;
    }

    public String getTitulo() {
        return titulo;
    }

    public String getAutor() {
        return autor;
    }

    public String getIsbn() {
        return isbn;
    }

    public String getEstado() {
        return estado;
    }

    public String getInfo() {
        return "El titulo del libro: " + getTitulo() + 
               "\nDel autor: " + getAutor() + 
               "\nCon ISBN: " + getIsbn() + 
               "\nTiene el estado: " + getEstado();
    }
}

class Biblioteca {
    private final ArrayList<Libro> libros;

    public Biblioteca() {
        this.libros = new ArrayList<>();
    }

    public void agregarLibro(Libro libro) {
        libros.add(libro);
    }

    public String buscarLibro(String isbn) {
        for (Libro libro : libros) {
            if (libro.getIsbn().equals(isbn)) {
                return libro.getInfo();
            }
        }
        return "No existe el libro";
    }

    public String prestarLibro(String isbn) {
        for (Libro libro : libros) {
            if (libro.getIsbn().equals(isbn)) {
                if (libro.getEstado().equals("Disponible")) {
                    libro.setEstado("Prestado");
                    return "Libro prestado con éxito";
                } else {
                    return "Este libro ya está prestado";
                }
            }
        }
        return "No existe ese libro";
    }

    public String devolverLibro(String isbn) {
        for (Libro libro : libros) {
            if (libro.getIsbn().equals(isbn)) {
                if (libro.getEstado().equals("Prestado")) {
                    libro.setEstado("Disponible");
                    return "Libro devuelto con éxito";
                } else {
                    return "Este libro ya está disponible";
                }
            }
        }
        return "No existe ese libro";
    }

    public String mostrarLibros() {
        if (libros.isEmpty()) {
            return "No hay libros en la biblioteca.";
        }

        StringBuilder infoLibros = new StringBuilder();
        for (Libro libro : libros) {
            infoLibros.append(libro.getInfo()).append("\n").append("-".repeat(30)).append("\n\n");
        }
        return infoLibros.toString();
    }

    public static void main(String[] args) {
        Libro l1 = new Libro("La divina comedia", "Dante Alighieri", "1", "Disponible");
        Libro l2 = new Libro("Don quijote de la mancha", "Miguel de Cervantes", "2", "Prestado");
        Libro l3 = new Libro("El paraíso perdido", "John Milton", "3", "Disponible");

        Biblioteca biblio = new Biblioteca();
        biblio.agregarLibro(l1);
        biblio.agregarLibro(l2);
        biblio.agregarLibro(l3);
        
        System.out.println(biblio.mostrarLibros());
        biblio.prestarLibro("1");
        System.out.println(biblio.mostrarLibros());
        biblio.devolverLibro("2");
        System.out.println(biblio.mostrarLibros());
    }
}